extensions/http/jersey/src/main/java/org/eclipse/dataspaceconnector/extension/jersey/JerseyRestService.java

```diff
@@ -83,8 +84,10 @@ private void registerContext(String contextAlias, List<Object> controllers) {
         resourceConfig.registerClasses(controllers.stream().map(Object::getClass).collect(toSet()));
         resourceConfig.registerInstances(new Binder());
         resourceConfig.registerInstances(new TypeManagerContextResolver(typeManager));
+
         resourceConfig.registerInstances(new EdcApiExceptionMapper());
```

Comment:
registering the exception mappers directly in the JerseyRestService will have two effects:
1. it will prevent users from implementing their own error handling. I imagine this may be problematic for some.
2. it will cause _all_ APIs to use these mappers, even quasi-internal ones, such as the Dataplane API, that are not meant for public consumption.

How about we register these three mappers in the `DataManagementApiConfigurationExtension` to make the JerseyRestService oblivious toward how errors need to be handled? That way the DM-API would still have proper error handling, but we don't force it on every other API.

Reply:
1. actually not, because every newly registered mapper will like be "chosen before" the other (with the same "distance" from the actual exception type), so everyone can register its own exception mapper
2. I think that's not a problem, as long as the "exception throwing" is correct that's a sane way to unify error behaviors (then every other module could have its own exception superclass and its own exception handler).

As this was already the used approach, I will propose to discuss this in another issue.

Reply:
ok, fine for me.